README.md

```diff
@@ -500,6 +502,75 @@ std::variant<fine::Ok<int64_t>, fine::Error<std::string>> find_meaning(ErlNifEnv
 Note that if you use a particular union frequently, it may be convenient
 to define a type alias with `using`/`typedef` to keep signatures brief.
 
+## Synchronization
+
+Erlang is a multi-process environment where each process is guaranteed to be
+isolated from other processes.  When dealing with NIFs, the same C++ function
+can be called from multiple Erlang processes simultaneously, leading to race
+conditions. While C++ provides synchronization mechanisms, these are unknown to
+Erlang and cannot take advantage of tools like *lock checker* or *lcnt*.
+
+Fine provides analogues to `std::mutex` and `std::shared_mutex`, respectively
+called `fine::Mutex` and `fine::RwLock`:
+
+```c++
+#include <fine/mutex.hpp>
+#include <fine/rwlock.hpp>
+
+fine::Mutex mutex;
+
+mutex.lock();
+mutex.unlock();
+bool locked = mutex.try_lock();
+
+{
+  auto lock = std::unique_lock(mutex);
+}
+
+fine::RwLock rwlock;
+
+rwlock.lock();
+rwlock.unlock();
+bool locked = rwlock.try_lock();
+
+rwlock.lock_shared();
+rwlock.unlock_shared();
+bool locked = rwlock.try_lock_shared();
+
+{
+  auto lock = std::shared_lock(rwlock);
+}
+
+{
+  auto lock = std::unique_lock(rwlock);
+}
+```
```

Comment:
Since they are analogous to the standard ones, I would simplify the example:

````suggestion
called `fine::Mutex` and `fine::RwLock`. Those are compatible with the standard
mutex wrappers, such as `std::unique_lock` and `std::shared_lock`. For example:

```c++
#include <fine/sync.hpp>

fine::Mutex mutex;

{
  auto lock = std::unique_lock(mutex);
  ...
}
```
````